packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -798,22 +796,42 @@ function FirstTermDiscountCallout( { product }: { product: ResponseCartProduct }
 		// Do not display discount reason if a coupon is applied.
 		isCouponApplied( product )
 	) {
-		return null;
+		return <></>;
 	}
-
+	let oneTimeExplanation: ReactChild = '';
 	if ( isMonthlyProduct( product ) ) {
-		return <DiscountCallout>{ translate( 'Discount for first month' ) }</DiscountCallout>;
+		oneTimeExplanation = translate( 'Discount: %(discount)s Applied in first month only', {
+			args: {
+				discount: formatCurrency( discount, product.currency, {
+					isSmallestUnit: true,
+					stripZeros: true,
+				} ),
+			},
+		} );
 	}
 
 	if ( isYearly( product ) ) {
-		return <DiscountCallout>{ translate( 'Discount for first year' ) }</DiscountCallout>;
+		oneTimeExplanation = translate( 'Discount: %(discount)s Applied in first year only', {
+			args: {
+				discount: formatCurrency( discount, product.currency, {
+					isSmallestUnit: true,
+					stripZeros: true,
+				} ),
+			},
+		} );
 	}
 
 	if ( isBiennially( product ) || isTriennially( product ) ) {
-		return <DiscountCallout>{ translate( 'Discount for first term' ) }</DiscountCallout>;
+		oneTimeExplanation = translate( 'Discount: %(discount)s Applied in first term only', {
```

Comment:
Could these assignments be changed back to early return statements? They are generally easier to reason about since the order is more clear (the first one that matches wins instead of the other way around).